src/map/script.cpp

```diff
@@ -26957,6 +26957,37 @@ BUILDIN_FUNC(has_autoloot) {
 	return SCRIPT_CMD_SUCCESS;
 }
 
+// ===================================
+// *autoloot(<rate>{, <char_id>});
+// This command sets the rate of autoloot (10000 = 100%).
+// Returns true on success and false on failure.
+// Note: It will toggle existing autoloot settings if rate value aren't provided.
+// ===================================
+BUILDIN_FUNC(autoloot) {
+	map_session_data *sd;
+
+	if (!script_charid2sd(3, sd)) {
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int rate = 0;
+	if (script_hasdata(st, 2))
+		rate = script_getnum(st, 2);
+	else if (sd->state.autoloot <= 0)
+		rate = 10000;
+
+	if (rate < 0 || rate > 10000) {
+		ShowWarning("buildin_autoloot: invalid rate value %d, shall be between 0 ~ 10000.\n", rate);
+		rate = cap_value(rate, 0, 10000);
```

Comment:
```suggestion
		script_pushint(st, false);
		return SCRIPT_CMD_FAILURE;
```

Reply:
In my opinion, if it doesn't cause major issues, it should be safe to continue execute the script. Capped the value, but the result are no different than what the user initially wanted, so it just throw a warning and cap it and proceed.

Reply:
This way the script being executed (path) gets reported to the user.
Otherwise you only get the warning message and do not know which script causes it.

And they should fix their scripts and not rely on being able to code badly.

Reply:
> This way the script being executed (path) gets reported to the user.
Otherwise you only get the warning message and do not know which script causes it.

I see, that's a good point.